components/patient-data/rest/src/test/java/org/phenotips/data/rest/internal/DefaultPatientResourceImplTest.java

```diff
@@ -0,0 +1,361 @@
+import org.xwiki.security.authorization.Right;
+import org.xwiki.test.mockito.MockitoComponentMockingRule;
+import org.xwiki.users.User;
+import org.xwiki.users.UserManager;
+
+import javax.inject.Provider;
+import javax.ws.rs.WebApplicationException;
+import javax.ws.rs.core.MediaType;
+import javax.ws.rs.core.Response;
+import javax.ws.rs.core.Response.Status;
+import javax.ws.rs.core.UriInfo;
+
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.util.List;
+import java.util.Map;
+
+import static org.hamcrest.Matchers.hasEntry;
+import static org.hamcrest.Matchers.hasItem;
+import static org.hamcrest.Matchers.hasValue;
+import static org.mockito.Matchers.any;
+import static org.mockito.Matchers.anyString;
+import static org.mockito.Matchers.eq;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.doReturn;
+import static org.mockito.Mockito.doThrow;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+/**
+ * Tests for the {@link DefaultPatientResourceImpl} component.
+ *
+ */
+public class DefaultPatientResourceImplTest {
+
+    @Rule
+    public MockitoComponentMockingRule<PatientResource> mocker =
+        new MockitoComponentMockingRule<PatientResource>(DefaultPatientResourceImpl.class);
+
+    @Mock
+    private Logger logger;
+
+    @Mock
+    private PatientRepository repository;
+
+    @Mock
+    private AuthorizationManager access;
+
+    @Mock
+    private UserManager users;
+
+    @Mock
+    private User currentUser;
+
+    @Mock
+    private Patient patient;
+
+    @Mock
+    private UriInfo uriInfo;
+
+    private URI uri;
+
+    private String uriString = "http://self/uri";
+
+    private DefaultPatientResourceImpl patientResource;
+
+    private String id = "00000001";
+
+
+    @Before
+    public void setUp() throws ComponentLookupException, URISyntaxException {
+        MockitoAnnotations.initMocks(this);
+        Execution execution = mock(Execution.class);
+        ExecutionContext executionContext = mock(ExecutionContext.class);
+
+        ComponentManager componentManager = this.mocker.getInstance(ComponentManager.class, "context");
+        when(componentManager.getInstance(Execution.class)).thenReturn(execution);
+        doReturn(executionContext).when(execution).getContext();
+        doReturn(mock(XWikiContext.class)).when(executionContext).getProperty("xwikicontext");
+
+        this.patientResource = (DefaultPatientResourceImpl)this.mocker.getComponentUnderTest();
+
+        doReturn(this.currentUser).when(this.users).getCurrentUser();
+        doReturn(null).when(this.currentUser).getProfileDocument();
+
+        doReturn(this.patient).when(this.repository).getPatientById(this.id);
+        doReturn(null).when(this.patient).getDocument();
+
+        this.uri = new URI(this.uriString);
+        doReturn(this.uri).when(this.uriInfo).getRequestUri();
+
+        ReflectionUtils.setFieldValue(this.patientResource, "logger", this.logger);
```

Comment:
Will this work for the Logger? Logger is an interface but it is not an `@Role`, and I'm not sure where it is getting the implementation, since there is no `@Named` hint. When I tried `this.mocker.getInstance(Logger.class)` a ComponentLookupException was thrown